### ISSUE TITLE
traefik in swarm mode does not work on windows

### ISSUE THREAD

**Issue by lindback** [traefik/traefik#7432]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Compose file with traefik 1.7.26 on windows works with docker-compose but not with docker swarm. October patches are applied so exposed ports in swarm are reachable (again).

### What did you expect to see?
When doing curl to port 80 (traefik endpoint), curl should return IIS page

### What did you see instead?
Curl just hang, access log show nothing. Running in docker-compose on windows works fine, on Linux docker-compose and swarm both works. (setting swarmmode=false when running compose of course)

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2020/10/19 11:28:32 Using high precision timer
Version:      v1.7.26
Codename:     maroilles
Go version:   go1.14.6
Built:        2020-07-28_03:45:27PM
OS/Arch:      windows/amd64 
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Windows version:
```
Windows 2019, 1809, Build 17763.1518 (mid October patches applied)
```
Docker version:
```
Server: Docker Engine - Enterprise (Unlicensed - not for production workloads)
 Engine:
  Version:          19.03.11
  API version:      1.40 (minimum version 1.24)
  Go version:       go1.13.11
  Git commit:       0da829ac52
  Built:            06/26/2020 17:19:32
  OS/Arch:          windows/amd64
  Experimental:     false
```
Compose file used:
```
version: '3.5'
services:
  web:
    image: mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019
    labels:
      - "traefik.port=80"
      - "traefik.frontend.rule=PathPrefix:/"
    deploy:
      labels:
        - "traefik.port=80"
        - "traefik.frontend.rule=PathPrefix:/"
    networks:
      - web

  loadbalancer:
    image: traefik:v1.7.26-windowsservercore-1809
    command: --docker
      --docker.swarmmode=true
      --docker.endpoint=npipe:////./pipe/docker_engine
      --loglevel=INFO
      --accesslog
      --entrypoints="Name:http Address::80 Compress:true"
      --defaultentrypoints="http"
    ports:
      - target: 80
        published: 80
        mode: host

    deploy:
      mode: global
      placement:
        constraints: [node.role == manager]
    volumes:
      - type: npipe
        source: \\.\pipe\docker_engine
        target: \\.\pipe\docker_engine
    networks:
      - web

networks:
  web:
    name: web
```
### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | 2020/10/19 11:30:48 Using high precision timer
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Traefik version v1.7.26 built on 2020-07-28_03:45:27PM"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":false,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false,\"AccessLogsFile\":\"\",\"AccessLog\":{\"format\":\"common\"},\"TraefikLogsFile\":\"\",\"TraefikLog\":null,\"Tracing\":null,\"LogLevel\":\"DEBUG\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":true,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"npipe:////./pipe/docker_engine\",\"Domain\":\"\",\"TLS\":null,\"ExposedByDefault\":true,\"UseBindPortIP\":false,\"SwarmMode\":true,\"Network\":\"\",\"SwarmModeRefreshSeconds\":15},\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":null,\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":null,\"Metrics\":null,\"Ping\":null,\"HostResolver\":null}"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/v1.7/basics/#collected-data\n"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:true ProxyProtocol:<nil> ForwardedHeaders:0xc0004b4040} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Starting server on :80"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Starting provider *docker.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"npipe:////./pipe/docker_engine\",\"Domain\":\"\",\"TLS\":null,\"ExposedByDefault\":true,\"UseBindPortIP\":false,\"SwarmMode\":true,\"Network\":\"\",\"SwarmModeRefreshSeconds\":15}"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Provider connection established with docker 19.03.11 (API 1.40)"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="originLabelsmap[com.docker.stack.image:traefik:v1.7.26-windowsservercore-1809 com.docker.stack.namespace:test]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="allLabelsmap[:map[]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Filtering container with empty frontend rule test_loadbalancer.vlaac31p4gwr0rh7560ez6bdj "
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Backend backend-test-web: no load-balancer defined, fallback to 'wrr' method"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Configuration received from provider docker: {\"backends\":{\"backend-test-web\":{\"servers\":{\"server-test-web-1-20f1f57f4df28b6bdb3ba8feff877fd8\":{\"url\":\"http://10.0.8.6:80\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"frontend-PathPrefix-0\":{\"entryPoints\":[\"http\"],\"backend\":\"backend-test-web\",\"routes\":{\"route-frontend-PathPrefix-0\":{\"rule\":\"PathPrefix:/\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Wiring frontend frontend-PathPrefix-0 to entryPoint http"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Creating backend backend-test-web"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Creating load-balancer wrr"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Creating server server-test-web-1-20f1f57f4df28b6bdb3ba8feff877fd8 at http://10.0.8.6:80 with weight 1"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=debug msg="Creating route route-frontend-PathPrefix-0 PathPrefix:/"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:30:48Z" level=info msg="Server configuration reloaded on :80"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="originLabelsmap[com.docker.stack.image:traefik:v1.7.26-windowsservercore-1809 com.docker.stack.namespace:test]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="allLabelsmap[:map[]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="Filtering container with empty frontend rule test_loadbalancer.vlaac31p4gwr0rh7560ez6bdj "
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="Backend backend-test-web: no load-balancer defined, fallback to 'wrr' method"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=debug msg="Configuration received from provider docker: {\"backends\":{\"backend-test-web\":{\"servers\":{\"server-test-web-1-20f1f57f4df28b6bdb3ba8feff877fd8\":{\"url\":\"http://10.0.8.6:80\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"frontend-PathPrefix-0\":{\"entryPoints\":[\"http\"],\"backend\":\"backend-test-web\",\"routes\":{\"route-frontend-PathPrefix-0\":{\"rule\":\"PathPrefix:/\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:03Z" level=info msg="Skipping same configuration for provider docker"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="originLabelsmap[com.docker.stack.image:traefik:v1.7.26-windowsservercore-1809 com.docker.stack.namespace:test]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="allLabelsmap[:map[]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="Filtering container with empty frontend rule test_loadbalancer.vlaac31p4gwr0rh7560ez6bdj "
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="originLabelsmap[com.docker.stack.image:mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019 com.docker.stack.namespace:test traefik.frontend.rule:PathPrefix:/ traefik.port:80]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="allLabelsmap[:map[traefik.frontend.rule:PathPrefix:/ traefik.port:80]]"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="Backend backend-test-web: no load-balancer defined, fallback to 'wrr' method"
test_loadbalancer.0.vlaac31p4gwr@usn1bvdockd04    | time="2020-10-19T11:31:18Z" level=debug msg="Configuration received from provider docker: {\"backends\":{\"backend-test-web\":{\"servers\":{\"server-test-web-1-20f1f57f4df28b6bdb3ba8feff877fd8\":{\"url\":\"http://10.0.8.6:80\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"frontend-PathPrefix-0\":{\"entryPoints\":[\"http\"],\"backend\":\"backend-test-web\",\"routes\":{\"route-frontend-PathPrefix-0\":{\"rule\":\"PathPrefix:/\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
```

**Comment by ddtmachado** [comment]
Hello @lindback , it seems that the root cause, as discussed on [this](https://github.com/moby/moby/issues/40998) docker issue, was fixed on [KB4580390 / OS Build 17763.1554](https://support.microsoft.com/en-us/topic/october-20-2020-kb4580390-os-build-17763-1554-preview-ac4799c9-838f-8665-a968-0f19b6cb1049) which seems to be a little more up to date than what's listed on the description.

Can you confirm if this issue remains for you?

**Comment by lindback** [comment]
> Hello @lindback , it seems that the root cause, as discussed on [this](https://github.com/moby/moby/issues/40998) docker issue, was fixed on [KB4580390 / OS Build 17763.1554](https://support.microsoft.com/en-us/topic/october-20-2020-kb4580390-os-build-17763-1554-preview-ac4799c9-838f-8665-a968-0f19b6cb1049) which seems to be a little more up to date than what's listed on the description.
> 
> Can you confirm if this issue remains for you?

No, but we found a work-around that we started using early on, by removing "mode=host" on ports, so I cannot really say if it still would be a problem with host mode or not.

**Comment by sdelicata** [comment]
Hello!
I close this issue as it targets an unsupported version of Traefik.
Feel free to re-open it or join our [Community Forum](https://community.traefik.io/).
